doc/internals/todo.md

```diff
@@ -0,0 +1,74 @@
+Todo for Source
+===============
+
+Case expressions and ifs
+------------------------
+
+We probably need slightly clunkier syntax than Haskell, unless we want to implement full-blown whitespace sensitivity.
+I think something like this would be fine.
+```
+option d v
+ = match v
+   | None   -> d
+   | Some a -> a
+   end
+```
```

Comment:
this looks pretty reasonable though, and does have the benefit that it's easily typed in to the REPL